core/eolearn/core/utilities.py

```diff
@@ -475,3 +476,113 @@ def constant_pad(X, multiple_of, up_down_rule='even', left_right_rule='even', pa
 def bgr_to_rgb(bgr):
     """Converts Blue, Green, Red to Red, Green, Blue."""
     return bgr[..., [2, 1, 0]]
+
+
+def map_over_axis(data, func, axis=0):
+    """Map function func over each slice along axis.
+    If func changes the number of dimensions, mapping axis is moved to the front.
+
+    Returns a new array with the combined results of mapping.
+
+
+    
+    :param data: input array
+    :type data: np.array
+    :param func: Mapping function that is applied on each slice. Outputs must have the same shape for every slice.
+    :type func: function np.array -> np.array
+    :param axis: Axis over which to map the function.
+    :type axis: int
+
+    :example:
+
+    >>> data = np.ones((5,10,10))
+    >>> func = lambda x: np.zeros((7,20))
+    >>> res = map_over_axis(data,func,axis=0)
+    >>> res.shape
+    (5, 7, 20)
+    """
+
+    # Move axis to front
+    data = np.moveaxis(data, axis, 0)
+    
+    res_mapped = [func(slice) for slice in data]
+    res = np.stack(res_mapped)
+    
+    # Move axis back if number of dimensions stays the same
+    if data.ndim == res.ndim:
+        res = np.moveaxis(res, 0, axis)
+    
+    return res
+
+def resize_images(data, new_size=None, scale_factors=None, anti_alias=True, interpolation='linear'):
+    """Resizes the image(s) acording to given size or scale factors.
+
+    To specify the new scale use one of `new_size` or `scale_factors` parameters.
+    
+    :param data: input image array
+    :type data: numpy array with shape (timestamps, height, width, channels), (height, width, channels), or (height, width)
+    :param new_size: New size of the data (width, height)
+    :type new_size: (int, int)
+    :param scale_factors: Factors (fx,fy) by which to resize the image
+    :type scale_factors: (float, float)
+    :param anti_alias: Use anti aliasing smoothing operation when downsampling.
+    :type anti_alias: bool
+    :param interpolation: Interpolation method used for resampling. One of 'nearest', 'linear', 'cubic'.
+    :type interpolation: bool
```

Comment:
typo: type of `interpolation` is `str` and not `bool`. It's also a good practice IMO to mention explicitly what the default value is for parameters that have one (like this one).